src/client/app/effects/snippet.ts

```diff
@@ -144,6 +144,29 @@ export class SnippetEffects {
             return Observable.of(new Snippet.LoadTemplatesSuccessAction([]));
         });
 
+    @Effect()
+    updateInfo$: Observable<Action> = this.actions$
+        .ofType(Snippet.SnippetActionTypes.UPDATE_INFO)
+        .map(( { payload } ) => {
+            let { id, name, description, gist, owned } = payload;
+            let snippet: ISnippet = storage.lastOpened;
+            if (storage.snippets.contains(id)) {
+                snippet = storage.snippets.get(id);
+
+                /* check if fields are undefined or null */
+                name != null ? snippet.name = name : snippet.name = snippet.name;
```

Comment:
snippet.name = snippet.name looks like a complicated no-op. Can we just change all of these
if (name != null) {
    snippet.name = name;
}